### PR TITLE
chore(layer): add license and architecture fields

### LINKED ISSUE #1371: Maintenance: add license info and compatible architecture to the public layer information
### Summary

The current info of public layer does not have `LicenseInfo` and `CompatibleArchitectures` fields. 

### Why is this needed?

Users might need this information for their internal approval if they can use the layer or not. In addition, there are tools that query `CompatibleArchitecture` field. 

### Which area does this relate to?

_No response_

### Solution

Add `license` and `compatibleArchitectures` field to the [layer-stack ](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/layers/src/layer-publisher-stack.ts#L31).

### Acknowledgment

- [X] This request meets [Lambda Powertools Tenets](https://awslabs.github.io/aws-lambda-powertools-typescript/latest/#tenets)
- [ ] Should this be considered in other Lambda Powertools languages? i.e. [Python](https://github.com/awslabs/aws-lambda-powertools-python/), [Java](https://github.com/awslabs/aws-lambda-powertools-java/), and [.NET](https://github.com/awslabs/aws-lambda-powertools-dotnet/)

### Future readers

Please react with 👍 and your use case to help us understand customer demand.

### PR DESCRIPTION
**Pull request by am29d**
## Description of your changes

This PR adds `LicenseInfo` and `CompatibleArchitectures` fields to the layer. 

### Related issues, RFCs

**Issue number:** closes #1371 

## Checklist

- [x] [My changes meet the tenets criteria](https://awslabs.github.io/aws-lambda-powertools-typescript/#tenets)
- [x] I have performed a *self-review* of my own code
- [x] I have *commented* my code where necessary, particularly in areas that should be flagged with a TODO, or hard-to-understand areas
- [ ] I have made corresponding changes to the *documentation*
- [ ] I have made corresponding changes to the *examples*
- [x] My changes generate *no new warnings*
- [x] The *code coverage* hasn't decreased
- [x] I have *added tests* that prove my change is effective and works
- [x] New and existing *unit tests pass* locally and in Github Actions
- [x] Any *dependent changes have been merged and published*
- [x] The PR title follows the [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/.github/semantic.yml#L2)

### Breaking change checklist

***Is it a breaking change?:*** NO

- [ ] I have documented the migration process
- [ ] I have added, implemented necessary warnings (if it can live side by side)

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

File: layers/src/layer-publisher-stack.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ import {
   LayerVersion,
   Code,
   Runtime,
-  CfnLayerVersionPermission
+  CfnLayerVersionPermission, Architecture
```
Comment (dreamorosi) [inline layers/src/layer-publisher-stack.ts:12]:
```suggestion
  CfnLayerVersionPermission,
  Architecture
```

File: layers/src/layer-publisher-stack.ts [hunk 2]
```diff
@@ -36,6 +36,8 @@ export class LayerPublisherStack extends Stack {
         Runtime.NODEJS_16_X,
         Runtime.NODEJS_18_X
       ],
+      license: 'MIT-0',
+      compatibleArchitectures: [Architecture.X86_64],
```
Comment (dreamorosi) [inline layers/src/layer-publisher-stack.ts:40]: What do you think what make us comfortable to add ARM here?

My understanding is that with the current dependency set and with the fact that this is Node.js, our packages should be supported for ARM without requiring any change.

I do however think that it'd be a good idea to test this. Do you think this should be something that is tested in the Layer integration tests, or in the beta deployment, or in the canary (that we want to add), or elsewhere?

**Review by dreamorosi — COMMENTED** [review]
Good one, left two small comments.

After that we can merge it.

File: layers/src/layer-publisher-stack.ts (hunk 2 above)
Comment (am29d) [reply layers/src/layer-publisher-stack.ts:40]: Adding ARM support would require a dedicated canary, in addition what we have planned for the layer publishing workflow #1355 . I'd address ARM support in a separate issue and PR. 

Do you want me to extend tests for license and architecture?

**Pull request merged**